Makefile

```diff
@@ -4,15 +4,18 @@ RISCV_OBJCOPY ?= $(CROSS_COMPILE)objcopy
 
 .PHONY: clean
 
-RISCV_CFLAGS ?= -march=rv64im -mabi=lp64 -mcmodel=medany -static -nostdlib -nostartfiles
-bootrom.riscv: cce_ucode.o
-	$(RISCV_GCC) $(RISCV_CFLAGS) bootrom.S $^ -I$(@D) -o $@ -Tlink.ld -static -Wl,--no-gc-sections
+RISCV_CFLAGS ?= -march=rv64imafd -mabi=lp64 -mcmodel=medany -static -nostdlib -nostartfiles
+bootrom.riscv: bootrom.S cce_ucode.bin
+	$(RISCV_GCC) -o $@ $(RISCV_CFLAGS) $< -I$(@D) -Tlink.ld -static -Wl,--no-gc-sections
 
 # Currently only one bootrom is generated at a time
 COH_PROTO ?= mesi
-cce_ucode.o:
-	$(RISCV_OBJCOPY) -I binary -O elf64-littleriscv -B riscv $(BP_SDK_UCODE_DIR)/$(COH_PROTO).bin $@ --strip-all --add-symbol cce_ucode_bin=.data:0
+
+cce_ucode.bin: $(BP_SDK_UCODE_DIR)/$(COH_PROTO).bin
+	$(RISCV_OBJCOPY) -I binary -O binary --reverse-bytes=8 $< $@
```

Comment:
why is byte reversal required here?

Reply:
Endianness is different between the binary and incbin. Not really sure which is the odd one, but the reversal clears it up

Reply:
makes sense. Might be the ucode. The LSB of the ucode is at the right hand side of the line, and I think the bytes of the file start with the left hand side of the first line.